jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/traveler/TravelerMessageController.java

```diff
@@ -1,56 +1,97 @@
+import us.dot.its.jpo.ode.snmp.SnmpProperties;
+import us.dot.its.jpo.ode.snmp.TimManagerService;
+import us.dot.its.jpo.ode.snmp.TimParameters;
 
 
 @Controller
 public class TravelerMessageController {
+    
+    private TravelerMessageController() {}
 
-
-    private static final String MESSAGE_DIR = "./src/test/resources/CVMessages/";
-
-    @RequestMapping(value = "/travelerMessage", method = RequestMethod.POST, produces = "application/json")
+    @RequestMapping(value = "/tim", method = RequestMethod.POST, produces = "application/json")
     @ResponseBody
-    public static String timMessage(@RequestBody String jsonString ) throws Exception {
+    public static String timMessage(@RequestBody String jsonString ) {
+        
+        // TODO Loop through all RSUs in the JSON and create and send an SNMP message to each
+        
+        Logger logger = LoggerFactory.getLogger(TravelerMessageController.class);
 
         if (jsonString == null) {
             throw new IllegalArgumentException("[ERROR] Endpoint received null TIM");
         }
-        JSONObject obj = new JSONObject(jsonString);
-//        String msgcnt = obj.getString("msgcnt");
-//
-//        // Standard Tim Message
-//        String timHex = "3081C68001108109000000000000003714830101A481AE3081AB800102A11BA119A0108004194FBA1F8104CE45CE2382020A0681020006820102820207DE830301C17084027D00850102A6108004194FC1988104CE45DA4082020A008702016E880100A92430228002000EA21CA01AA31804040CE205A104040ADA04F70404068004D60404034D0704AA3AA0383006A004800235293006A0048002010C3006A004800231283006A004800222113006A0048002010C3006A004800231203006A0048002221185021001";
-//        byte [] tim_ba = HexTool.parseHex(timHex, false);
-//        InputStream ins = new ByteArrayInputStream(tim_ba);
-//
-//        TravelerInformation tim = new TravelerInformation();
-//
-//        System.out.print(tim);
-//        try {
-//            coder.decode(ins, tim);
-//
-//        } catch (Exception e) {
-////            System.out.print( e);
-//        } finally {
-//            try {
-//                ins.close();
-//            } catch (IOException e) {
-//            }
-//        }
-//
-//        MsgCount cnt = new MsgCount(msgcnt);
-//        tim.setMsgCnt(cnt);
+        
+        
+        // Step 1 - Serialize the JSON into a TIM object
         TravelerSerializer timObject = new TravelerSerializer(jsonString);
-        System.out.print(timObject.getTravelerInformationObject());
+        
+        
+        // Step 2 - Populate the SnmpProperties object with SNMP preferences
+        JSONObject obj = new JSONObject(jsonString);
+        JSONArray rsuList = obj.getJSONArray("RSUs");
+        String ip = rsuList.getJSONObject(0).getString("target");
+        String user = rsuList.getJSONObject(0).getString("username");
+        String pass = rsuList.getJSONObject(0).getString("password");
+        int retries = Integer.parseInt(rsuList.getJSONObject(0).getString("retries"));
```

Comment:
Please use JSONObject's getInt method instead of getString method and then parsing it to integer.